src/QEQ/fix_qeq_ctip.cpp

```diff
@@ -0,0 +1,394 @@
+#define A4 -1.453152027
+#define A5 1.061405429
+
+/* ---------------------------------------------------------------------- */
+
+FixQEqCTIP::FixQEqCTIP(LAMMPS *lmp, int narg, char **arg) :
+  FixQEq(lmp, narg, arg) {
+
+  cdamp = 0.30;
+  maxrepeat = 10;
+
+  // optional arg
+  int iarg = 8;
+  while (iarg < narg) {
+    if (strcmp(arg[iarg], "cdamp") == 0) {
+      if (iarg + 2 > narg) utils::missing_cmd_args(FLERR, "fix qeq/ctip cdamp", error);
+      cdamp = utils::numeric(FLERR, arg[iarg + 1], false, lmp);
+      iarg += 2;
+    } else if (strcmp(arg[iarg], "maxrepeat") == 0) {
+      if (iarg + 2 > narg) utils::missing_cmd_args(FLERR, "fix qeq/ctip maxrepeat", error);
+      maxrepeat = utils::numeric(FLERR, arg[iarg + 1], false, lmp);
+      iarg += 2;
+    } else if (strcmp(arg[iarg], "warn") == 0) {
+      if (iarg + 2 > narg) utils::missing_cmd_args(FLERR, "fix qeq/ctip warn", error);
+      maxwarn = utils::logical(FLERR, arg[iarg + 1], false, lmp);
+      iarg += 2;
+    } else
+      error->all(FLERR, "Unknown fix qeq/ctip keyword: {}", arg[iarg]);
+  }
+
+  if (ctip_flag) extract_ctip();
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixQEqCTIP::init()
+{
+  FixQEq::init();
+
+  neighbor->add_request(this, NeighConst::REQ_FULL);
+
+  int ntypes = atom->ntypes;
+  memory->create(shld,ntypes+1,ntypes+1,"qeq:shielding");
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixQEqCTIP::extract_ctip()
+{
+  Pair *pair = force->pair_match("coul/ctip",1);
+  if (pair == nullptr) error->all(FLERR,"No pair coul/ctip for fix qeq/ctip");
+  int tmp;
+  chi = (double *) pair->extract("chi",tmp);
+  eta = (double *) pair->extract("eta",tmp);
+  gamma = (double *) pair->extract("gamma",tmp);
+  zeta = (double *) pair->extract("zeta",tmp);
+  zcore = (double *) pair->extract("zcore",tmp);
+  qmin = (double *) pair->extract("qmin",tmp);
+  qmax = (double *) pair->extract("qmax",tmp);
+  omega = (double *) pair->extract("omega",tmp);
+  if (chi == nullptr || eta == nullptr || gamma == nullptr
+                  || zeta == nullptr || zcore == nullptr || qmin == nullptr || qmax == nullptr || omega == nullptr)
+    error->all(FLERR,
+        "Fix qeq/ctip could not extract params from pair coul/ctip");
+
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixQEqCTIP::pre_force(int /*vflag*/)
+{
+
+  int i,n;
+
+  if (update->ntimestep % nevery) return;
+
+  nlocal = atom->nlocal;
+
+  if (atom->nmax > nmax) reallocate_storage();
+
+  if (nlocal > n_cap*DANGER_ZONE || m_fill > m_cap*DANGER_ZONE)
+    reallocate_matrix();
+
+  for (i=1; i <= maxrepeat; i++) {
+       init_matvec();
+       matvecs = CG(b_s, s);         // CG on s - parallel
+       matvecs += CG(b_t, t);        // CG on t - parallel
+       matvecs /= 2;
+       n=calculate_check_Q();
+       MPI_Allreduce(&n, &nout, 1, MPI_INTEGER, MPI_SUM, world);
```

Comment:
MPI_INTEGER is a Fortran MPI interface constant. In C/C++ this would be MPI_INT.

```suggestion
       MPI_Allreduce(&n, &nout, 1, MPI_INT, MPI_SUM, world);
```